22.Semaphore/main.go

```diff
@@ -49,7 +52,11 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
-	ch := make(chan int, 2)
+	n, err := strconv.Atoi(os.Args[1])
+	if err != nil {
+		log.Fatal(err)
+	}
+	ch := make(chan int, n)
```

Comment:
why you use chan int? I recommend to use chan struct{} in such cases (when value not important, and channel is used for lock/unlock)